lib/src/modules/task.cc

```diff
@@ -373,7 +385,7 @@ static fs::path updateTaskFile(const std::vector<std::string>& master_uris,
                                lth_curl::client& client,
                                const fs::path& cache_dir,
                                const lth_jc::JsonContainer& file) {
-    auto filename = file.get<std::string>("filename");
+    auto filename = fs::basename(fs::path(file.get<std::string>("filename")));
```

Comment:
We chose to keep the cache dir "flat" and name files based on their base name. This will overwrite duplicate files but the complexity of maintaining a nested cache dir is not worth the minor performance issue of re-downloading and overwriting cached files.

Reply:
It shouldn't overwrite duplicate files. If the file has the same hash, we shouldn't redownload it in the first place.

Reply:
The `if` block at line 380/392 will take care of that.